File: source/kl/render/Vertex.java
package kl.render;

import kl.math.*;

public class Vertex {
    public Float3 world;
    public Float2 texture;
    public Float3 normal;

    public Vertex() {
        world = new Float3();
        texture = new Float2();
        normal = new Float3();
    }
    public Vertex(Float3 world) {
        this.world = world;
        texture = new Float2();
        normal = new Float3();
    }
    public Vertex(Float3 world, Float2 texture) {
        this.world = world;
        this.texture = texture;
        normal = new Float3();
    }
    public Vertex(Float3 world, Float2 texture, Float3 normal) {
        this.world = world;
        this.texture = texture;
        this.normal = normal;
    }
    public Vertex(Vertex v) {
        world = new Float3(v.world);
        texture = new Float2(v.texture);
        normal = new Float3(v.normal);
    }

    public String toString() {
        return "{" + world + ", " + texture + ", " + normal + "}";
    }
}

File: source/render/vertex.cpp
#include "render/vertex.h"


kl::vertex::vertex()
{}

kl::vertex::vertex(const float3& world)
    : world(world)
{}

kl::vertex::vertex(const float3& world, const float2& texture)
    : world(world), texture(texture)
{}

kl::vertex::vertex(const float3& world, const float3& normal)
    : world(world), normal(normal)
{}

kl::vertex::vertex(const float3& world, const float2& texture, const float3& normal)
    : world(world), texture(texture), normal(normal)
{}

std::ostream& kl::operator<<(std::ostream& stream, const vertex& obj)
{
    stream << "{" << obj.world << ", " << obj.texture << ", " << obj.normal << "}";
    return stream;
}
